### ISSUE TITLE
Core dumped

### ISSUE THREAD

**Issue by swvanderlaan** [qtltools/qtltools#10]
Hi,

I've been trying to run QTLTools, however I get a 'core dumped' error message at the start of the analysis. I'm not sure where to look for a solution. Can you give me some advice?

Thanks,

Sander

**Command:**
```
QTLtools_1.0_CentOS6.8_x86_64 cis --vcf aegs_1kGp3GoNL5_QC_cg02757179_excl_EXCL_FEMALES.vcf.gz --bed /hpc/dhl_ec/data/_ae_originals/AEMS450K1/20180302.AEMS450K1.BvaluesQCplaque.QuickFix.bed.gz --region 02:24384809-26384809 --window 2000000 --nominal 0.05 --exclude-samples exclusion_nonCEA_Females.list --exclude-covariates excl_cov.txt --cov covariates_aegs_combo.all.cov --out aems450k1_QC_qtlnom_cg02757179_excl_EXCL_FEMALES.txt.gz --log aems450k1_QC_qtlnom_cg02757179_excl_EXCL_FEMALES.log
```

**Error message (in bold):**

```
QTLtools
  * Authors : Olivier DELANEAU / Halit ONGEN / Emmanouil DERMITZAKIS
  * Contact : olivier.delaneau@gmail.com / halit.ongen@unige.ch / Emmanouil.Dermitzakis@unige.ch
  * Webpage : https://qtltools.github.io/qtltools/
  * Version : 1.0
  * Date    : 12/03/2018 - 22:59:35

MAPPING QTL IN CIS
  * TASK: Report all nominal associations with p <= 0.05
  * Linear model: Phe ~ Var + Cov
  * Cis-window size is 2000000 bp
  * Region = [02:24384809-26384809]

Read sample exclusion list [/hpc/dhl_ec/data/_ae_originals/AEGS_COMBINED_IMPUTE2_1000Gp3_GoNL5/exclusion_nonCEA_Females.list]
  * 547 samples

Read covariate exclusion list [/hpc/dhl_ec/svanderlaan/projects/epigenetics/autosomal_sex/excl_cov.txt]
  * 66 covariates

Initialize random number generator
  * Built-in seed is 15112011
  * First Integer = 30554
  * First Double = 0.0126695

Reading sample list from [/hpc/dhl_ec/data/_ae_originals/AEMS450K1/20180302.AEMS450K1.BvaluesQCplaque.QuickFix.bed.gz]
  * #samples included by user = 305
  * #samples excluded by user = 139

Read sample list from [/hpc/dhl_ec/svanderlaan/projects/epigenetics/autosomal_sex/aems450k1_malesonly_small/EXCL_FEMALES_qtl/cg02757179_aems450k1_malesonly_small/aegs_1kGp3GoNL5_QC_cg02757179_excl_EXCL_FEMALES.vcf.gz]
  * #samples included by user = 979
  * #samples excluded by user = 547

Reading sample list from [/hpc/dhl_ec/data/_ae_originals/AEGS_COMBINED_IMPUTE2_1000Gp3_GoNL5/covariates_aegs_combo.all.cov]
  * #samples included by user = 979
  * #samples excluded by user = 547

Merging sample sets from all input files
  * #files = 3
  * #samples in all files = 305
  * #samples NOT in all files (i.e. ignored) = 674

Reading phenotype data in [/hpc/dhl_ec/data/_ae_originals/AEMS450K1/20180302.AEMS450K1.BvaluesQCplaque.QuickFix.bed.gz]
  * target region [02:24384809-26384809]
  * 430 phenotypes included

Reading genotype data in [/hpc/dhl_ec/svanderlaan/projects/epigenetics/autosomal_sex/aems450k1_malesonly_small/EXCL_FEMALES_qtl/cg02757179_aems450k1_malesonly_small/aegs_1kGp3GoNL5_QC_cg02757179_excl_EXCL_FEMALES.vcf.gz]
  * File format detected: VCF
  * target region [02:22384809-28384809]
  * 3187 variants included

Reading covariates in [/hpc/dhl_ec/data/_ae_originals/AEGS_COMBINED_IMPUTE2_1000Gp3_GoNL5/covariates_aegs_combo.all.cov]
  * 7 covariates included
  * 66 covariates excluded

Residualize phenotypes for covariates
**terminate called after throwing an instance of 'std::invalid_argument'
  what():  stof
Aborted (core dumped)**
```

**Comment by swvanderlaan** [comment]
Hi,

I googled the error - and found 'stof' might have to do with the _size_ of the numbers. By that I mean, whether the numbers are smaller than 1. I want to do a methylation QTL (mQTL) analysis and so my values are anywhere between 0 and 1 (beta-values). Could that be the issue? I could use the M-values instead - would that help?

Thanks and best,

Sander

**Comment by swvanderlaan** [comment]
Ah, yes. Is this relevant: [https://stackoverflow.com/questions/34108016/validating-floating-point-in-c-without-special-libraries](https://stackoverflow.com/questions/34108016/validating-floating-point-in-c-without-special-libraries)?

**Comment by clemgoub** [comment]
I had this error because I had "NA" in my covariates file

**Comment by halitongen** [comment]
This was due to a bug in how missing values were handled. They should be marked as NA, and this bug is fixed in version 1.2.
